### PR TITLE
Model compression progress bar

### PR DESCRIPTION
**Pull request by Rezmason**
## Summary
The model transform function now reports its progress, and the compression panel now has a captioned progress bar instead of a spinner.

## References
closes https://tsu.atlassian.net/browse/IR-4271

### CODE REVIEW

**Review by dinomut1 — DISMISSED** [review]
After running a default compression job on a file in "ir-props", the integrated prefab file is getting uploaded into /assets/assets/models rather than /assets/models
image

**Comment by Rezmason** [comment]
> After running a default compression job on a file in "ir-props", the integrated prefab file is getting uploaded into /assets/assets/models rather than /assets/models

image

File: packages/editor/src/components/assets/ModelCompressionPanel.tsx [hunk 1]
```diff
@@ -51,17 +54,32 @@ import { SourceComponent } from '@ir-engine/engine/src/scene/components/SourceCo
...
 import GLTFTransformProperties from '../properties/GLTFTransformProperties'
 
+const progressCaptions: Map<ModelTransformStatus, string> = new Map([
+  [ModelTransformStatus.Initializing, 'editor:properties.model.transform.status.initializing'],
+  [ModelTransformStatus.ProcessingTexture, 'editor:properties.model.transform.status.processingtexture'],
+  [ModelTransformStatus.Finalizing, 'editor:properties.model.transform.status.finalizing'],
+  [ModelTransformStatus.WritingFiles, 'editor:properties.model.transform.status.writingfiles'],
+  [ModelTransformStatus.Complete, 'editor:properties.model.transform.status.complete']
+])
+
```
Comment (aditya-mitra) [inline packages/editor/src/components/assets/ModelCompressionPanel.tsx]: in this case, the key value pairs are known. so an plain object would work here. we should utilise a map when we are dynamically adding or removing key values.

File: packages/engine/src/assets/compression/ModelTransformFunctions.ts [hunk 2]
```diff
@@ -343,9 +343,24 @@ function hashBuffer(buffer: Uint8Array): string {
...
+export { Status as ModelTransformStatus }
+
 export async function transformModel(
   args: ModelTransformParameters,
-  onMetadata: (key: string, data: any) => void = (key, data) => {}
+  onMetadata: (key: string, data: any) => void = (key, data) => {},
+  onProgress: (progress: number, status: Status, numerator?: number, denominator?: number) => void = (
+    numerator,
+    denominator,
+    status
```
Comment (aditya-mitra) [inline packages/engine/src/assets/compression/ModelTransformFunctions.ts]:
```suggestion
    _numerator,
    _denominator,
    _status
```

we can use `_` underscore to denote unused variable in this case

**Review by aditya-mitra — DISMISSED** [review]
few changes

File: packages/engine/src/assets/compression/ModelTransformFunctions.ts (hunk 2 above)
Comment (Rezmason) [reply packages/engine/src/assets/compression/ModelTransformFunctions.ts]: These are in a function declaration that's a default value for a function parameter, so I'll just delete the default value and make the function parameter optional.

Believe it or not, I think the formatter wrote these out. I was using `() => {}` as the default value and I guess the formatter didn't like my arg erasure. It's a funny old world

File: packages/editor/src/components/assets/ModelCompressionPanel.tsx (hunk 1 above)
Comment (Rezmason) [reply packages/editor/src/components/assets/ModelCompressionPanel.tsx]: TIL that the TypeScript Record type with an enum key is exhaustive, ie. it throws a useful compiler error if there are any missing enum keys in the declaration ✨

**Comment by Rezmason** [comment]
> After running a default compression job on a file in "ir-props", the integrated prefab file is getting uploaded into /assets/assets/models rather than /assets/models 

Found to be caused by https://github.com/ir-engine/ir-engine/pull/78

**Review by dinomut1 — APPROVED** [review]

**Pull request merged**